presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -2559,6 +2603,27 @@ public void revokeTablePrivileges(ConnectorSession session, SchemaTableName sche
         return toCompletableFuture(stagingFileCommitter.commitFiles(session, handle.getSchemaName(), handle.getTableName(), getPartitionUpdates(fragments)));
     }
 
+    private HivePartitioningHandle createHivePartitioningHandle(
+            int bucketCount,
+            HiveBucketHandle hiveBucketHandle,
+            Optional<HiveBucketProperty> bucketProperty,
+            OptionalInt maxCompatibleBucketCount)
+    {
+        if (bucketProperty.isPresent() &&
+                bucketProperty.get().getBucketFunctionType().equals(PRESTO_NATIVE)) {
```

Comment:
Since bucketProperty could be empty,  this function is a little different from other places:  if hive bucker property exists and indicates to use `PRESTO_NATIVE`, then use it, otherwise,  use the old way.

Reply:
It feels like the bucketProperty should be empty only if the table is virtually bucketed. Let's have a special case for that.